packages/client/src/__tests__/integration/happy/interactive-transactions/test.ts

```diff
@@ -0,0 +1,410 @@
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_2@website.com',
+        },
+      }),
+    ])
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(2)
+  })
+
+  /**
+   * A bad batch should rollback using the interactive transaction logic
+   */
+  test('batching rollback', async () => {
+    const result = prisma.$transaction([
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+    ])
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+Invalid \`prisma.user.create()\` invocation:
+
+
+  Unique constraint failed on the fields: (\`email\`)
+`)
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(0)
+  })
+
+  /**
+   * Middlewares should work normally on batches
+   */
+  test('middlewares batching', async () => {
+    prisma.$use(async (params, next) => {
+      const result = await next(params)
+
+      return result
+    })
+
+    await prisma.$transaction([
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_2@website.com',
+        },
+      }),
+    ])
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(2)
+  })
+
+  /**
+   * Middlewares should not prevent a rollback
+   */
+  test('middlewares batching rollback', async () => {
+    prisma.$use(async (params, next) => {
+      const result = await next(params)
+
+      return result
+    })
+
+    const result = prisma.$transaction([
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+      prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      }),
+    ])
+
+    await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+
+Invalid \`prisma.user.create()\` invocation:
+
+
+  Unique constraint failed on the fields: (\`email\`)
+`)
+
+    const users = await prisma.user.findMany()
+
+    expect(users.length).toBe(0)
+  })
+
+  /**
+   * Minimal example of a interactive transaction & middleware
+   */
+  test('middleware basic', async () => {
+    prisma.$use(async (params, next) => {
+      await next(params)
+
+      return 'result'
+    })
+
+    const result = await prisma.$transaction((prisma) => {
+      return prisma.user.create({
+        data: {
+          email: 'user_1@website.com',
+        },
+      })
+    })
+
+    expect(result).toBe('result')
+  })
+})
+
+beforeAll(async () => {
+  PrismaClient = await getTestClient()
+
+  process.env.PRISMA_FORCE_LRT = 'true'
+})
+
+afterAll(() => {
+  delete process.env.PRISMA_FORCE_LRT
```

Comment:
I could not find where `PRISMA_FORCE_LRT` logic is.

Also is it for internal testing only?

If we need to keep it I would recommend updating the name because LRT change to "interactive transactions"

Reply:
Good catch, thanks!